packages/vite-plugin-angular/src/lib/authoring/analog.ts

```diff
@@ -77,7 +78,12 @@ export function compileAnalogFile(
   const componentMetadata = (() => {
     if (ngType === 'Component') {
       const items = ['changeDetection: ChangeDetectionStrategy.OnPush'];
-      if (templateContent) {
+
+      if (fileContent.includes('lang="md"')) {
```

Comment:
suggestion: I'd compute this flag ahead and check if this is true then skip the `TEMPLATE_TAG_REGEX.exec()` in this file.